src/olcs/FeatureConverter.js

```diff
@@ -495,16 +517,63 @@ exports.prototype.olPolygonGeometryToCesium = function(layer, feature, olGeometr
       perPositionHeight: true
     });
 
-    outlineGeometry = new Cesium.PolygonOutlineGeometry({
-      // always update Cesium externs before adding a property
-      polygonHierarchy: hierarchy,
-      perPositionHeight: true
-    });
+    // Since Cesium doesn't yet support Polygon outlines on terrain yet (coming soon...?)
+    // we don't create an outline geometry if clamped, but instead do the polyline method
+    // for each ring. Most of this code should be removeable when Cesium adds
+    // support for Polygon outlines on terrain.
+    if (heightReference == Cesium.HeightReference.CLAMP_TO_GROUND) {
+      const width = this.extractLineWidthFromOlStyle(olStyle);
+      if (width > 0) {
+        const positions = [hierarchy.positions];
+        if (hierarchy.holes) {
+          for (let i = 0; i < hierarchy.holes.length; ++i) {
+            positions.push(hierarchy.holes[i].positions);
+          }
+        }
+        if (!Cesium.GroundPolylinePrimitive.isSupported(this.scene)) {
+          const color = this.extractColorFromOlStyle(olStyle, true);
+          outlineGeometry = this.createStackedGroundCorridors(width, color, positions);
+        } else {
+          const appearance = new Cesium.PolylineMaterialAppearance({
+            // always update Cesium externs before adding a property
+            material: this.olStyleToCesium(feature, olStyle, true)
+          });
+          const primitiveOptions = {
+            // always update Cesium externs before adding a property
+            appearance
+          };
```

Comment:
Now that only one primitive is created you should move the object creation after the loop.